src/lib/config/index.ts

```diff
@@ -279,16 +279,51 @@ const readFileOpt = (fileName: string) => {
   }
 };
 
-const getRootConfigFile = () => {
+const readFile = (fileName: string) => {
+  return fs.readFileSync(fileName, "utf8");
+};
+
+const getRootConfigPath = () => {
   return path.join(os.homedir(), ".fauna-shell");
 };
 
-// TODO: Search upwards for a `.fauna-project` file
-const getProjectConfigPath = () => {
-  const projectConfigPath = path.join(process.cwd(), ".fauna-project");
-  if (fs.existsSync(projectConfigPath)) {
-    return projectConfigPath;
-  } else {
+export const getProjectConfigPath = (): string | undefined => {
+  let current;
+  try {
+    current = process.cwd();
+  } catch (err) {
+    // If the cwd is not accessible, just give up. If this happens, one of our
+    // dependencies actually explodes elsewhere, but we might as well handle
+    // errors where possible.
     return undefined;
   }
+
+  for (let i = 0; i < 2048; i++) {
```

Comment:
Does it need to be a bounded search? By definition, every filesystem I'm aware of will hit the `.base === ""` case eventually.

Reply:
it doesn't need to be bounded. I was thinking it might be possible to create a loop with symlinks or something, but I can just make it unbounded.